test/helpers/log-exploration-api.go

```diff
@@ -0,0 +1,218 @@
+package helpers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	clolog "github.com/ViaQ/logerr/log"
+	"github.com/openshift/cluster-logging-operator/internal/factory"
+	"github.com/openshift/cluster-logging-operator/internal/k8shandler"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"log"
+	"strconv"
+	"time"
+)
+
+const maxLogsLimit = 5
+
+type logExplorationAPI struct {
+	deployment *apps.Deployment
+	service    *corev1.Service
+	tc         *E2ETestFramework
+}
+
+func (l *logExplorationAPI) ApplicationLogs(timeToWait time.Duration) (types.Logs, error) {
+	panic("implement me")
+}
+func (l *logExplorationAPI) HasApplicationLogs(timeToWait time.Duration) (bool, error) {
+
+	logs, err := l.getLogs()
+	if err != nil {
+		return false, err
+	}
+	if len(logs) <= 1 {
+		time.Sleep(180 * time.Second)
+		logs, err = l.getLogs()
+		if err == nil && len(logs) > 1 {
+			return true, nil
+		} else {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+func (l *logExplorationAPI) getLogs() ([]interface{}, error) {
+	options := metav1.ListOptions{
+		LabelSelector: "component=elasticsearch",
+	}
+
+	pods, err := l.tc.KubeClient.CoreV1().Pods(OpenshiftLoggingNS).List(context.TODO(), options)
+	if err != nil {
+		return nil, err
+	}
+	if len(pods.Items) == 0 {
+		return nil, errors.New("No pods found for elasticsearch")
+	}
+	clolog.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+	stdout, err := l.tc.PodExec(OpenshiftLoggingNS, pods.Items[0].Name, "elasticsearch", []string{"curl", "http://logexplorationapi:8080/logs?maxlogs=" + strconv.Itoa(maxLogsLimit)})
```

Comment:
Why exec from an ES pod? Cant you do this same query from the logexploration pod?